### PR TITLE
Add association between raw tracker/calorimeter hit and tracker/calorimeter hit

### PR DESCRIPTION
**Pull request by rahmans1**
### Briefly, what does this PR introduce?
Add association between raw tracker hit and tracker hit to have full association path from MCParticle to tracker hit. 

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### CODE REVIEW

File: edm4eic.yaml [hunk 1]
```diff
@@ -259,6 +259,8 @@ datatypes:
       - int32_t           sector            // Sector that this hit occurred in
       - int32_t           layer             // Layer that the hit occurred in
       - edm4hep::Vector3f local             // The local coordinates of the hit in the detector segment [mm]. 
+    OneToOneRelations:
+      - edm4eic::RawCalorimeterHit rawHit       // Related raw calorimeter hit
```
Comment (wdconinc) [inline edm4eic.yaml:263]:
```suggestion
      - edm4eic::RawCalorimeterHit rawHit   // Related raw calorimeter hit
```

File: edm4eic.yaml [hunk 2]
```diff
@@ -558,3 +562,12 @@ datatypes:
     OneToOneRelations:
      - edm4eic::RawTrackerHit rawHit        // reference to the digitized hit
      - edm4hep::SimTrackerHit simHit       // reference to the simulated hit
```
Comment (wdconinc) [inline edm4eic.yaml:564]:
```suggestion
     - edm4hep::SimTrackerHit simHit        // reference to the simulated hit
```

File: edm4eic.yaml [hunk 3]
```diff
@@ -558,3 +562,12 @@ datatypes:
...
+
+  edm4eic::MCRecoCalorimeterHitAssociation:
+    Description: "Association between a RawCalorimeterHit and a SimCalorimeterHit"
+    Author: "C. Dilks, W. Deconinck"
+    Members:
+      - float                 weight        // weight of this association
+    OneToOneRelations:
+     - edm4eic::RawCalorimeterHit rawHit        // reference to the digitized calorimeter hit
+     - edm4hep::SimCalorimeterHit simHit        // reference to the simulated calorimeter hit
```
Comment (wdconinc) [inline edm4eic.yaml:573]:
```suggestion
     - edm4eic::RawCalorimeterHit rawHit    // reference to the digitized calorimeter hit
     - edm4hep::SimCalorimeterHit simHit    // reference to the simulated calorimeter hit
```

File: edm4eic.yaml [hunk 4]
```diff
@@ -558,3 +562,12 @@ datatypes:
     OneToOneRelations:
      - edm4eic::RawTrackerHit rawHit        // reference to the digitized hit
      - edm4hep::SimTrackerHit simHit       // reference to the simulated hit
+
+  edm4eic::MCRecoCalorimeterHitAssociation:
+    Description: "Association between a RawCalorimeterHit and a SimCalorimeterHit"
+    Author: "C. Dilks, W. Deconinck"
```
Comment (wdconinc) [inline edm4eic.yaml:568]:
```suggestion
    Author: "S. Rahman"
```

**Review by wdconinc — DISMISSED** [review]
Looks good! I think this adds the remaining links at the input of the tracker and calorimeter chains.

We discussed this yesterday and we think there is no need for a weight in the TrackerHit and CalorimeterHit, nor is there a need for a OneToManyRelations in these data types.

**Review by wdconinc — APPROVED** [review]

**Review by veprbl — APPROVED** [review]
This was discussed quite extensively previously in reco meetings. The change looks good to me.

**Review by ruse-traveler — DISMISSED** [review]
Very nice!

**Review by wdconinc — APPROVED** [review]
Backwards compatible change.

**Pull request merged**